### PR TITLE
OSDOCS#7442: Agent data gathering

### PR DESCRIPTION
**Pull request by skopacz1**
[OSDOCS-7442](https://issues.redhat.com/browse/OSDOCS-7442)

Versions: 4.13+

This PR adds instructions for gathering data for a support case when you Agent-based installation has failed.

QE review:
- [x] QE has approved this change.

Preview: [Gathering log data from a failed Agent-based installation](https://64183--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_with_agent_based_installer/installing-with-agent-based-installer#installing-ocp-agent-gather-log_installing-with-agent-based-installer)

### CODE REVIEW

File: modules/installing-ocp-agent-gather-log.adoc [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+----
+//to-do: check whether the must gather command needs any specific arguments in order to gather data specific to the agent install. The KCS this procedure is based on says to "Use the kubeconfig file available on <installation directory>/auth." but I don't know what that means in this case.
+
+.. Create a compressed file from the `must-gather` directory that was just created in your working directory. For example, on a computer that uses a Linux operating system, run the following command:
++
+[source,terminal]
+----
+$ tar cvaf must-gather.tar.gz must-gather.local.5421342344627712289/ <1>
+----
+<1> Replace `must-gather.local.5421342344627712289/` with the actual directory name.
```
Comment (skopacz1) [inline modules/installing-ocp-agent-gather-log.adoc:55]: I took these steps from the [standard must-gather procedure](https://docs.openshift.com/container-platform/4.13/support/gathering-cluster-data.html#support_gathering_data_gathering-cluster-data), let me know if this needs any adjustment for the agent-specific use case.

File: modules/installing-ocp-agent-gather-log.adoc [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+[source,terminal]
+----
+$ ./openshift-install --dir <install_directory> agent wait-for bootstrap-complete --log-level=debug
+----
+
+. If the output from the previous command indicates a failure, or if the bootstrap is not progressing, run the following command for each node and collect its output:
++
+[source,terminal]
+----
+$ ssh core@<node-ip> sudo /usr/local/bin/agent-gather -O > <local_tmp_path>/agent-gather.tar.xz
```
Comment (skopacz1) [inline modules/installing-ocp-agent-gather-log.adoc:24]: If you were to gather this data from each node, then all you would change is the <node-ip> value above each time, right?

Comment (bfournie) [reply modules/installing-ocp-agent-gather-log.adoc:24]: Yes

File: modules/installing-ocp-agent-gather-log.adoc [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+.. Create a compressed file from the `must-gather` directory that was just created in your working directory. For example, on a computer that uses a Linux operating system, run the following command:
++
+[source,terminal]
+----
+$ tar cvaf must-gather.tar.gz must-gather.local.5421342344627712289/ <1>
+----
+<1> Replace `must-gather.local.5421342344627712289/` with the actual directory name.
+
+. Excluding the `kubeconfig` file, attach the installation directory used during the deployment to your support case on the link:https://access.redhat.com[Red Hat Customer Portal].
```
Comment (wking) [inline modules/installing-ocp-agent-gather-log.adoc:57]: You probably want to exclude the whole `auth/` subdirectory, because [it also contains a `kubeadmin-password` file][1].

[1]: https://github.com/openshift/installer/blob/75ac0821ee012d8855dadf42c25cc807d8ef8d51/cmd/openshift-install/create.go#L615

File: modules/installing-ocp-agent-gather-log.adoc [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+. If the output from the previous command indicates a failure, perform the following steps:
+
+.. Run the following command:
++
+[source,terminal]
+----
+$ oc adm must-gather
+----
+//to-do: check whether the must gather command needs any specific arguments in order to gather data specific to the agent install. The KCS this procedure is based on says to "Use the kubeconfig file available on <installation directory>/auth." but I don't know what that means in this case.
```
Comment (patrickdillon) [inline modules/installing-ocp-agent-gather-log.adoc:47]: No special arguments would be needed, but if an agent install does not save the kubeconfig to this same path you may want to describe where to grab the kubeconfig.

Comment (bfournie) [reply modules/installing-ocp-agent-gather-log.adoc:47]: Yes the agent installer does save the kubeconfig to this path.

Comment (skopacz1) [reply modules/installing-ocp-agent-gather-log.adoc:47]: Just confirmed with Bob, it is saved to the same path. Thanks!

**Comment by skopacz1** [comment]
@mhanss could you PTAL when you have the chance? Thanks!

File: modules/installing-ocp-agent-gather-log.adoc [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+.. Create a compressed file from the `must-gather` directory that was just created in your working directory. For example, on a computer that uses a Linux operating system, run the following command:
++
+[source,terminal]
+----
+$ tar cvaf must-gather.tar.gz must-gather.local.5421342344627712289/ <1>
```
Comment (ShaunaDiaz) [inline modules/installing-ocp-agent-gather-log.adoc:60]:
```suggestion
$ tar cvaf <must-gather.tar.gz must-gather.local.5421342344627712289/> <1>
```
Afiak we use the <value> to indicate a user-inserted value.

**Review by ShaunaDiaz — APPROVED** [review]
Looks great! A few questions and suggestions for your consideration.

File: modules/installing-ocp-agent-gather-log.adoc [hunk 6]
```diff
@@ -0,0 +1,66 @@
+// Module included in the following assemblies:
+//
+// * installing-with-agent/installing-with-agent.adoc
```
[the remark this answers is not in the document]
Comment (skopacz1) [reply modules/installing-ocp-agent-gather-log.adoc:3]: Good catch, I think I copied this over from a module that had an outdated include statement. I'll fix them all.

File: modules/installing-ocp-agent-gather-log.adoc [hunk 7]
```diff
@@ -0,0 +1,64 @@
+// Module included in the following assemblies:
+//
+// * installing/installing-with-agent-based-installer/installing-with-agent-based-installer.adoc
+
+:_content-type: PROCEDURE
+[id="installing-ocp-agent-gather-log_{context}"]
+= Gathering log data from a failed Agent-based installation
+
+Use the following procedure to gather log data about a failed Agent-based installation to provide for a support case.
```
Comment (kelbrown20) [inline modules/installing-ocp-agent-gather-log.adoc:9]: Just a question, does this procedure need any prereqs. Its not always necessary, but I wanted to ask to make sure.

File: modules/installing-ocp-agent-gather-log.adoc [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+.Procedure
+
+. Run the following command and collect its output:
```
Comment (kelbrown20) [inline modules/installing-ocp-agent-gather-log.adoc:13]: I think the `its` part needs a bit more description. Maybe like 
`Collect the output of a failed Agent-based installation`
^^
This might not be technically correct but I think you get what I mean 😄 

Also, not a huge deal, but usually we'd want to write the description of the command, then add the "by running to by entering the following command" part
[Shown mostly here in the doc guidelines  ](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc?plain=1#L1048)

File: modules/installing-ocp-agent-gather-log.adoc [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
++
+[source,terminal]
+----
+$ ./openshift-install --dir <install_directory> agent wait-for bootstrap-complete --log-level=debug
+----
+
+. If the output from the previous command indicates a failure, or if the bootstrap is not progressing, run the following command on node 0 and collect its output:
```
Comment (kelbrown20) [inline modules/installing-ocp-agent-gather-log.adoc:20]: Do you think its worth adding an example output of what a failed or successful collection of an output? Or even maybe a brief description of it? I can see doing that might over crowd the procedure cuz I'm not sure what a failed or successful collection looks like, but I was curious about your thoughts as well.

**Review by kelbrown20 — COMMENTED** [review]
Hi Seb! I just had some nits and could questions to consider before merging. Great work on this!

File: modules/installing-ocp-agent-gather-log.adoc (hunk 7 above)
Comment (skopacz1) [reply modules/installing-ocp-agent-gather-log.adoc:9]: I don't think it does. 

I'm assuming that users who got this far have satisfied the prereqs to perform an install in the first place. And step 4a (Export the kubeconfig file to your environment by running the following command) allows you to use the oc CLI, otherwise I believe the procedure would require the common "access to the cluster as an administrator" prereq.

File: modules/installing-ocp-agent-gather-log.adoc (hunk 8 above)
Comment (skopacz1) [reply modules/installing-ocp-agent-gather-log.adoc:13]: Regarding "its" needing more description: "it" refers to the command, so the instruction is to run the command and copy whatever output it gives you. I could reword it to something like "Run the following command and collect the output of the command" but imo I think it's a bit too wordy.

Regarding the "by running the following command" structure: The guideline says that structure should be followed in most cases, however in these steps, running the command is the first of two actions taken in the step (running the command and then collecting the output of the command). I'd have to split the step into two steps to follow what's really a guideline, whereas this format still follows the requirement to make sure there's an explicit instruction to "run" the command.

Is it fine to keep this step (and the others like it) the way it is? Let me know what you think!

File: modules/installing-ocp-agent-gather-log.adoc (hunk 9 above)
Comment (skopacz1) [reply modules/installing-ocp-agent-gather-log.adoc:20]: It might be worth it, I'll ask an SME to see if there's a short and easy example to mention. It's tough because this is sort of an open-ended procedure, where likely any number of things can indicate a failure, and maybe it's buried in a lot of output.

But if there's an easy example to mention from the SME, I can try and include it here.

File: modules/installing-ocp-agent-gather-log.adoc (hunk 8 above)
Comment (kelbrown20) [reply modules/installing-ocp-agent-gather-log.adoc:13]: Ohhh okay! I wasn't correctly reading what I thought the command was doing. And yeah that makes sense for the format of the procedure referring to the "by running" part. I'm still wondering, what do you think of just, 
`Run the following command and collect the output`
I personally feel like this ^^ is a bit simpler and conveys the same thing. But thats also a bit nitty so its not a huge deal if you want to keep what you have 😄

File: modules/installing-ocp-agent-gather-log.adoc (hunk 7 above)
Comment (kelbrown20) [reply modules/installing-ocp-agent-gather-log.adoc:9]: Gotcha! Yeah prereqs are not always necessary, and I know sometimes they get missed. So I mainly just wanted to ask. 👍

File: modules/installing-ocp-agent-gather-log.adoc (hunk 9 above)
Comment (kelbrown20) [reply modules/installing-ocp-agent-gather-log.adoc:20]: Yeah I've had that issue before with huge potential output. I've sometimes tried to put breifish (1-2 sentences) descriptions if the example output is too crazy. Even if there is something to show for a successful build can also be helpful bc users can have some kind of reference to go off of. Users can also probably tell if the command failed by comparing their output to the successful example, or vise versa. I hope that helps!

File: modules/installing-ocp-agent-gather-log.adoc (hunk 8 above)
Comment (skopacz1) [reply modules/installing-ocp-agent-gather-log.adoc:13]: Sure thing, that wording works for me

File: modules/installing-ocp-agent-gather-log.adoc (hunk 9 above)
Comment (skopacz1) [reply modules/installing-ocp-agent-gather-log.adoc]: I was able to get an example of an error message and added it to the previous step, where you run the command that would give you the message in the first place.

**Review by kelbrown20 — COMMENTED** [review]
Everything LGTM! Great work!
